### DISCUSSION TITLE
Running the Wazuh agent on same host as server

### DISCUSSION THREAD

**Discussion by cyb3rz3us** [Security-Onion-Solutions/securityonion#7680] [Unsupported Versions]
### Update-1 since OP
 
Based on more research, it appears that two scenarios exist WRT to installing the agent on the same host as the manager.  Some environments have the actual manager also performing the agent function to monitor the localhost yet others require the separate installation of the agent if one wants to monitor the host.  My SO install is clearly the latter.

Here the weird thing: if I manually stop the agent, it restarts every 14 minutes like clockwork.  I have checked cron, the container, etc. and I  cannot locate the mechanism by which the agent restarts.

Any ideas here?
- - -
### Original Post

Is is acceptable to run both the Wazuh agent and server on the same host?  The reason I ask is because at least 2 settings - both having to do with how often Wazuh performs various checks - do not behave as expected.

For example, the Wazuh documentation says the rootcheck execution interval can be modified via the ```<frequency>``` setting in the ```/var/ossec/etc/ossec.conf``` yet regardless of the value I use, it always runs on an 11-13 minute interval.  And the FIM ```<frequency>``` setting in that file doesn't appear to have any effect whatsoever.  I had to tune the FIM interval in the   ```/nsm/wazuh/etc/ossec.conf``` file.

I also read in a previous discussion that Salt is not used to control any Wazuh configuration but I have observed different behavior in my environment.  Whenever the ```so-wazuh-restart``` command is run, it appears that the ```/var/ossec/etc/ossec.conf``` file is overwritten with a file generated using what I think is a template found at ```/opt/so/saltstack/default/salt/wazuh/files/agent/ossec.conf```.

If running the Wazuh agent and server on the same machine is allowed, it would be great to get some clarification on the above.  Thanks in advance for your feedback.

**Comment by dougburks** [comment]
> Here the weird thing: if I manually stop the agent, it restarts every 14 minutes like clockwork. I have checked cron, the container, etc. and I cannot locate the mechanism by which the agent restarts.

Salt runs every 15 minutes to ensure the system is in the correct state:
https://docs.securityonion.net/en/2.3/salt.html
